src/tools/registry/gen_inc.c

```diff
@@ -83,6 +109,51 @@ int write_field_pointer_arrays(FILE* fd){/*{{{*/
 	return 0;
 }/*}}}*/
 
+// Checks for a string in a list of strings.
+// Returns the index of the string if it does exist in the array,
+// and -1 if it does not appear in the array.
+int find_string_in_array(char *input_string, const char *array[], size_t rows){
+	size_t i;
+	for (i = 0; i < rows; i++ ){
+		if (strcmp(input_string, array[i]) == 0){
+			return i;
+		}
+	}
+	return -1;
+}
+
+// Helper function to change attribute names in accordance with
+// "attrs_to_modify" within the "add_attribute_if_not_ignored" function
+const char * modify_attr(const char *attr, const char *array[][2], size_t rows) {
+	size_t i;
+	for (i = 0; i < rows; i++) {
+		if (strcmp(attr, array[i][0]) == 0) {
+			return array[i][1];
+		}
+	}
+	return attr;
+}
+
+void add_attribute_if_not_ignored(FILE *fd, char *index, char *att_name, char *pointer_name_arr, char *att_value){	
+	char *format_string;
+	
+	// Return early if we want to ignore the attribute
+	if (find_string_in_array(att_name, ATTRS_TO_IGNORE, NUM_IGNORED_ATTRS) >= 0){
+		return;
+	}
+
+	// check if the attribute is numeric
+	if (find_string_in_array(att_name, NUMERIC_ATTRS, NUM_NUMERIC_ATTRS) >= 0){
+		format_string = "      call mpas_add_att(%s %% attLists(%s) %% attList, \"%s\", %s)\n";
+	} 
+	// If it isn't numeric, make sure to wrap  att_value in quotes
+	else {
+ 		format_string = "      call mpas_add_att(%s %% attLists(%s) %% attList, \"%s\", \"%s\")\n";
```

Comment:
It looks like we're getting some compilation issues with the `init_atmosphere` core when attributes are long enough that the line splitting in the `fortprintf` should be acting upon a long code line.

Here's an example warning message:
```
./inc/structs_and_variables.inc:576:0:

  576 | velocity for an edge")
      | 
Warning: Missing ‘&’ in continued character constant at (1) [-Wampersand]
```

I think these all stem from the use of `"` characters as the string delimiters in the generated `mpas_add_att` calls in the `add_attribute_if_not_ignored` function.